code/datums/components/goliath_plate.dm

```diff
@@ -0,0 +1,30 @@
+/datum/component/goliath_plate
+	var/amount = 0
+
+/datum/component/goliath_plate/Initialize(obj/item/src,mob/user)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	RegisterSignal(COMSIG_PARENT_EXAMINE, .proc/examine)
+	RegisterSignal(COMSIG_PARENT_ATTACKBY, .proc/applyplate)
+	applyplate(src,user) //need to call it once since the component does not start applied to the armors
+
+/datum/component/goliath_plate/proc/examine(mob/user)
+	if(amount)
+		to_chat(user, "It has been strengthened with [amount] goliath plate\s.")
+	else
+		to_chat(user, "It can be strengthened with goliath hide plates.")
+
+/datum/component/goliath_plate/proc/applyplate(obj/item/I, mob/user)
+	if(!istype(I,/obj/item/stack/sheet/animalhide/goliath_hide))
```

Comment:
That wouldn't be goliath plate anymore. Also obviates need for passing these arguments to Initialize() because then it would be better to add component to each item in its Initialize() instead of adding it from goliath_hide's afterattack.

Reply:
I'm not sure what you're trying to say. Yes you'd add a var to determine the type that would come here.

The Goliath Plate's afterattack is for the explicit purpose of loading the components (not that I like that system but I didn't look much into it).. The component itself is SUPPOSE to handle everything or else it's useless making it a component as you've gained 0 abstraction.